client.go

```diff
@@ -1510,6 +1521,24 @@ func (c *Client[TTx]) insertManyShared(
 	execute func(context.Context, []*riverdriver.JobInsertFastParams) ([]*rivertype.JobInsertResult, error),
 ) ([]*rivertype.JobInsertResult, error) {
 	doInner := func(ctx context.Context) ([]*rivertype.JobInsertResult, error) {
+		for _, params := range insertParams {
+			// TODO(brandur): This range clause and the one below it are
+			// identical, and it'd be nice to merge them together, but in such a
+			// way that doesn't require array allocation. I think we can do this
+			// using iterators after we drop support for Go 1.22.
```

Comment:
Maybe this isn't worth being the one thing that pushes us to fully drop 1.22 support, but we're well within our "2 most recent Go versions" policy to support only 1.23+.

Reply:
Yeah true. Didn't quite feel like I should mess with the Go version for this, but I'm sure something will end up bumping it intentionally or not pretty soon, so I'll take a look then.